src/parser.cpp

```diff
@@ -1125,4 +1157,216 @@ void Parser::ParserImpl::loadImport(const ImportSourcePtr &importSource, const M
     }
 }
 
+void Parser::ParserImpl::loadReset(const ResetPtr &reset, const ComponentPtr &component, const XmlNodePtr &node)
+{
+    int order = 0;
+    bool orderDefined = false;
+    VariablePtr referencedVariable = nullptr;
+    std::string variableName = "";
+
+    XmlAttributePtr attribute = node->getFirstAttribute();
+    while (attribute) {
+        if (attribute->isType("variable")) {
+            const std::string variableReference = attribute->getValue();
+            referencedVariable = component->getVariable(variableReference);
+            if (referencedVariable == nullptr) {
+                ErrorPtr err = std::make_shared<Error>();
+                err->setDescription("Reset referencing variable '" + variableReference +
+                                    "' is not a valid reference for a variable in component '" + component->getName() + "'.");
+                err->setReset(reset);
+                err->setKind(Error::Kind::RESET);
+                err->setRule(SpecificationRule::RESET_INVALID_VARIABLE_REFERENCE);
+                mParser->addError(err);
```

Comment:
I am not sure I like this, i.e. call `setReset()` and then `setKind()`. I mean, we could, in principle, have a call `setKind()`, but then forget to call `setReset()`. In fact, if at all possible, I would have a simple call to `mParser->addError(...);`, to which all the parameters needed to create and customise `ErrorPtr` would be passed. We could, for example, pass `reset`, but not `Error::Kind::RESET` since this would be implied. This means that we would have several signatures for `addError()`, but at least there wouldn't be any risk for errors from a library user. Just a thought...

Reply:
I have done this in a later pull request (for validation #227) where I have simplified adding errors slightly.  I haven't gone the whole way on this but this can be added easily enough.

Reply:
Fair enough, this is not related to this PR, so it can be simplified some other time, I agree.